packages/lib-sourcify/src/Verification/Verification.ts

```diff
@@ -0,0 +1,475 @@
+    if (
+      !forceEmscripten && // Enter this case only if we are not already forcing Emscripten
+      lt(this.compilation.compilerVersion, '0.8.21') &&
+      !settings.optimizer?.enabled &&
+      settings.viaIR
+    ) {
+      logInfo('Force Emscripten compiler', {
+        address: this.address,
+        chainId: this.sourcifyChain.chainId,
+      });
+
+      // Try to verify again with Emscripten
+      return SolidityBugType.IR_OUTPUT_ORDERING_BUG;
+    }
+
+    // Case when extra unused files in compiler input cause different bytecode
+    // See issues:
+    //   https://github.com/ethereum/sourcify/issues/618
+    //   https://github.com/ethereum/solidity/issues/14250
+    //   https://github.com/ethereum/solidity/issues/14494
+    const [, deployedAuxdata] = splitAuxdata(
+      this.onchainRuntimeBytecode,
+      AuxdataStyle.SOLIDITY,
+    );
+    const [, recompiledAuxdata] = splitAuxdata(
+      compiledRuntimeBytecode,
+      AuxdataStyle.SOLIDITY,
+    );
+    // Metadata hashes match but bytecodes don't match.
+    if (deployedAuxdata === recompiledAuxdata && settings.optimizer?.enabled) {
+      throw new VerificationError(
+        "It seems your contract's metadata hashes match but not the bytecodes. If you are verifying via metadata.json, use the original full standard JSON input file that has all files including those not needed by this contract. See the issue for more information: https://github.com/ethereum/sourcify/issues/618",
+        'EXTRA_FILE_INPUT_BUG',
+      );
+    }
+
+    return SolidityBugType.NONE;
+  }
+
+  private async matchBytecodes(
+    isCreation: boolean,
+    normalizedRecompiledBytecode: string,
+  ): Promise<BytecodeMatchingResult> {
+    // Here we use bytecodes from the context because they are already processed
+    const onchainBytecode = isCreation
+      ? this.onchainCreationBytecode
+      : this.onchainRuntimeBytecode;
+
+    const cborAuxdata = isCreation
+      ? this.compilation.creationBytecodeCborAuxdata
+      : this.compilation.runtimeBytecodeCborAuxdata;
+
+    const transformations = isCreation
+      ? this.creationTransformations
+      : this.runtimeTransformations;
+    const transformationValues = isCreation
+      ? this.creationTransformationValues
+      : this.runtimeTransformationValues;
+
+    const linkReferences = isCreation
+      ? this.compilation.creationLinkReferences
+      : this.compilation.runtimeLinkReferences;
+
+    const result: BytecodeMatchingResult = {
+      match: null,
+      transformations: [...transformations],
+      transformationValues: { ...transformationValues },
+      normalizedRecompiledBytecode,
+    };
+
+    // Replace library placeholders
+    const librariesTransformationResult = extractLibrariesTransformation(
+      normalizedRecompiledBytecode,
+      onchainBytecode,
+      linkReferences,
+    );
+    const libraryMap = librariesTransformationResult.libraryMap;
+    normalizedRecompiledBytecode =
```

Comment:
Same comment as `extractCallProtectionTransformation` and immutables transformation function regarding "normalized" vs "populated". Adding for completeness